### PR TITLE
feat: reassign custom dimensions

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR set the custom dimensions properly so they get through to GA.

### CODE REVIEW

File: client/src/redux/ga-saga.js [hunk 1]
```diff
@@ -23,14 +23,19 @@ function* callGaType({ payload: { type, data } }) {
       const email = yield select(emailSelector);
       const completedChallengeTotal = yield select(completedChallengesSelector);
       const completedChallengeSession = yield select(completionCountSelector);
-      const customDimensions = {
-        Test_Variation: emailToABVariant(email).isAVariant ? 'A' : 'B',
-        Test_Type: aBTestConfig.type,
-        Challenges_Completed_Session: completedChallengeSession,
-        Challenges_Completed_Total: completedChallengeTotal.length,
-        URL: window.location.href
+      const CustomDimensions = {
```
Comment (ShaunSHamilton) [inline client/src/redux/ga-saga.js:26]: There is nothing special about this object from a JS point of view. So, I suggest sticking with naming convention

```suggestion
      const customDimensions = {
```

File: client/src/redux/ga-saga.js [hunk 2]
```diff
@@ -23,14 +23,19 @@ function* callGaType({ payload: { type, data } }) {
...
+        dimension2: completedChallengeSession,
+        // Challenges_Completed_Total
+        dimension3: completedChallengeTotal.length,
+        // Test_Type
+        dimension4: aBTestConfig.type,
+        // Test_Variation
+        dimension5: emailToABVariant(email).isAVariant ? 'A' : 'B'
       };
-      data = { ...data, ...customDimensions };
+      ga.set(CustomDimensions);
```
Comment (ShaunSHamilton) [inline client/src/redux/ga-saga.js:38]:
```suggestion
      ga.set(customDimensions);
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
If you would not mind explaining what the purpose of this PR is? Is it just handy to have the dimensions named as such on the analytics page?

**Review by ojeytonwilliams — DISMISSED** [review]
Shaun's suggestion aside, this LGTM 👍

**Comment by ahmadabdolsaheb** [comment]
This PR set the custom dimensions properly so they get through to GA.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by ojeytonwilliams — APPROVED** [review]

**Pull request merged**